packages/optimizer-docker/metrics.js

```diff
@@ -0,0 +1,38 @@
+const http = require('http');
+const url = require('url');
+
+const client = require('prom-client');
+
+const registry = new client.Registry();
+
+// Enable the collection of default metrics
+client.collectDefaultMetrics({register: registry});
+
+const mTransformedDurationGauge = new client.Summary({
+  name: 'transformer_duration_seconds',
+  help: 'Duration of the transformer in seconds',
+  percentiles: [0.5, 0.75, 0.9, 0.99],
+  labelNames: ['checkpoint'],
+  maxAgeSeconds: 120,
+  ageBuckets: 5,
+});
+
+registry.registerMetric(mTransformedDurationGauge);
+
+const profiler = function (label) {
+  return mTransformedDurationGauge.startTimer({checkpoint: label})
+};
```

Comment:
Do you have any reason for creating an anonymous fn and assigning it to this const vs. using regular named function definition? In particular I prefer top level items to be defined via named functions because of how it removes any ordering issues

```suggestion
function profiler(label) {
  return mTransformedDurationGauge.startTimer({checkpoint: label})
};
```

Reply:
no reason, this is mainly due to my poor js skills